### PR TITLE
Display `envtool run test` progress with run and/or skip flags

### LINKED ISSUE #3969: Process `--run` and `--skip` in the `envtool` itself
### What should be done?

Currently, when `envtool tests run` is invoked with `--run` and/or `--skip` flags, we just pass those values to `go test`'s `-run` and `-skip`, skip tests listing and sharding. That makes the output less useful: there is no known number of total tests, the progress indicator is worse, etc.

Let's get a list of tests as usual, then apply `--run` and `--skip`'s regular expressions ourselves, then shard the resulting list.

### Where?

https://github.com/FerretDB/FerretDB/blob/1e736b9d4a52fee51e6121c94fd048e3a136a1fb/cmd/envtool/tests.go#L356-L398

### Definition of Done

- `envtool` code updated;
- `envtool`'s unit tests added/updated;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by fadyat**
# Description

`envtool tests run` now display progress and the number of remaining tests when passing flags `--run` and/or `--skip` 

Closes #3969.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

**Review by henvic — APPROVED** [review]

File: cmd/envtool/tests.go [hunk 1]
```diff
@@ -448,6 +433,71 @@ func listTestFuncs(dir string) ([]string, error) {
...
+
+	excludeRegex, err := regexp.Compile(skip)
+	if err != nil {
+		return nil, err
+	}
+
+	return filterStringsByRegex(tests, includeRegex, excludeRegex), nil
+}
+
+func filterStringsByRegex(tests []string, include, exclude *regexp.Regexp) []string {
```
Comment (chilagrow) [inline cmd/envtool/tests.go:466]: Could we have documentation on unexported functions as well :hugs:  https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#code-style-and-conventions

**Review by chilagrow — COMMENTED** [review]
Great work, a tiny comment about adding documentation.

File: cmd/envtool/tests.go (hunk 1 above)
Comment (noisersup) [reply cmd/envtool/tests.go:468]: Looks resolved to me!

**Review by chilagrow — APPROVED** [review]
Amazing work, thanks for your contribution :hugs:

**Review by noisersup — APPROVED** [review]

**Pull request merged**